magic_bridge/ic/src/dip20_proxy/src/common/types.rs

```diff
@@ -78,7 +81,7 @@ pub struct ProxyState {
     /// store incoming messages against status locks
     pub incoming_messages: RefCell<HashMap<MessageHash, MessageStatus>>,
     /// user balances
-    pub balances: RefCell<HashMap<Principal, HashMap<TokendId, Nat>>>,
+    pub balances: RefCell<HashMap<Principal, HashMap<TokendId, Vec<(Principal, Nat)>>>>,
```

Comment:
Same idea as the `eth_proxy` balances structure.

- Simplify
- and if caller is not needed truncate to:

```rust
EthereumAddr
RefCell<HashMap<EthereumAddr, HashMap<TokenId, , Nat>>>,
```